### ISSUE TITLE
Add support for composer.json when using truncate_with_package_name

### ISSUE THREAD

**Issue by ettoredn** [bhilburn/powerlevel9k#349]
It would be nice to be able to truncate directory names with the name of a composer/PHP project. The strategy should use the name present in either package.json or composer.json depending on which one is present. When both are present, allow the user to choose which one has precedence.

Any suggestions?

**Comment by bhilburn** [comment]
I'm not a JS developer myself, so I don't think I have good enough of an understanding of normal workflows to weigh in, here.

A sane default will need to be chosen, though, which users can override - I'm thinking `package.json` since

**Comment by dritter** [comment]
@ettoredn Interesting idea! I actually use composer a lot. I'll see what I can do here, once I have some time (might be after #344 is done).

Btw. a quick idea regarding having a `composer.json` and a `package.json` in the same folder: I think that could be separate segments. [Here](https://github.com/bhilburn/powerlevel9k/compare/next...BenoitAverty:next) somebody implemented truncation with a stopfile. We could use that in combination with a separate composer/package.json segment. By joining (and some icon-tricks) it could look like one segment..

**Comment by dritter** [comment]
@ettoredn Have a look at #412 . There I implemented truncate via composer.json ;)

**Comment by bhilburn** [comment]
Another feature request knocked out by @dritter!

#412 has been merged into `next`!
